test/unit/math/rev/mat/prob/normal_id_glm_lpdf_gpu_test.cpp

```diff
@@ -0,0 +1,243 @@
+#ifdef STAN_OPENCL
+#include <stan/math.hpp>
+#include <stan/math/opencl/normal_id_glm_lpdf.hpp>
+#include <gtest/gtest.h>
+
+using Eigen::Array;
+using Eigen::Dynamic;
+using Eigen::Matrix;
+using stan::math::matrix_cl;
+using stan::math::var;
+
+TEST(ProbDistributionsNormalIdGLM, error_checking) {
+  double eps = 1e-9;
+  int N = 3;
+  int M = 2;
+
+  Matrix<double, Dynamic, 1> y(N, 1);
+  y << 51, 32, 12;
+  Matrix<double, Dynamic, 1> y_size(N + 1, 1);
+  y_size << 51, 32, 12, 34;
+  Matrix<double, Dynamic, 1> y_value(N, 1);
+  y_value << 51, 32, INFINITY;
+  Matrix<double, Dynamic, Dynamic> x(N, M);
+  x << -12, 46, -42, 24, 25, 27;
+  Matrix<double, Dynamic, Dynamic> x_size1(N - 1, M);
+  x_size1 << -12, 46, -42, 24;
+  Matrix<double, Dynamic, Dynamic> x_size2(N, M - 1);
+  x_size2 << -12, 46, -42;
+  Matrix<double, Dynamic, Dynamic> x_value(N, M);
+  x_value << -12, 46, -42, 24, 25, -INFINITY;
+  Matrix<double, Dynamic, 1> beta(M, 1);
+  beta << 0.3, 2;
+  Matrix<double, Dynamic, 1> beta_size(M + 1, 1);
+  beta_size << 0.3, 2, 0.4;
+  Matrix<double, Dynamic, 1> beta_value(M, 1);
+  beta_value << 0.3, INFINITY;
+  Matrix<double, Dynamic, 1> alpha(N, 1);
+  alpha << 0.3, -0.8, 1.8;
+  Matrix<double, Dynamic, 1> alpha_size(N - 1, 1);
+  alpha_size << 0.3, -0.8;
+  Matrix<double, Dynamic, 1> alpha_value(N, 1);
+  alpha_value << 0.3, -0.8, NAN;
+  Matrix<double, Dynamic, 1> sigma(N, 1);
+  sigma << 5, 2, 3.4;
+  Matrix<double, Dynamic, 1> sigma_size(N + 1, 1);
+  sigma_size << 5, 2, 3.4, 6;
+  Matrix<double, Dynamic, 1> sigma_value(N, 1);
+  sigma_value << 5, 2, NAN;
+
+  matrix_cl<double> x_cl(x);
+  matrix_cl<double> x_size1_cl(x_size1);
+  matrix_cl<double> x_size2_cl(x_size2);
+  matrix_cl<double> x_value_cl(x_value);
+  matrix_cl<double> y_cl(y);
+  matrix_cl<double> y_size_cl(y_size);
+  matrix_cl<double> y_value_cl(y_value);
+
+  EXPECT_NO_THROW(
+      stan::math::normal_id_glm_lpdf(y_cl, x_cl, alpha, beta, sigma));
```

Comment:
Also add tests that check the behavior for size zero. I dont see an equivalent of 
```
if (size_zero(y, x, beta, sigma)) {	  if (size_zero(y, x, beta, sigma)) {
    return 0;	    return 0;
  }
```
in the gpu glm, so I am guessing it doesnt work for that case? I did go and check that none of the GLMs have size zero tests which is not great. I will add an issue.

Reply:
In most functions this is just an optimization for special case. However an opencl kernel can not be launched with global size of 0, so I'm adding such check and tests.